Deprecate the Qt UI and its installers

davemfish opened the pull request:
Fixes #755 . See the checklist on that issue for all the items included in these changes.

## Checklist
- [x] Updated HISTORY.rst and link to any relevant issue (if these changes are user-facing)
- [ ] Updated the user's guide (if needed)
- [ ] Tested the affected models' UIs (if relevant)

davemfish commented:
@phargogh , don't mind me, just removing ~13,000 lines of your code. (Doug volunteered to review this one, but feel free 😃 )

dcdenu4 reviewed (changes requested):
Thanks @davemfish!

A few questions of whether these should also be removed:
- `ci/windows-ci-binary-install.ps1`
- `ci/mac_mini_build_process.sh`
- `doc\api-docs\cli.rst` has mention of `invest launch`
- `doc\api-docs\conf.py` Qt section referencing `InvestModelInput`?
- Do we need `exe/qt.conf` anymore?

Otherwise this is looking good!

davemfish commented:
> Thanks @davemfish!
> 
> A few questions of whether these should also be removed:
> 
> * `ci/windows-ci-binary-install.ps1`
> * `ci/mac_mini_build_process.sh`
> * `doc\api-docs\cli.rst` has mention of `invest launch`
> * `doc\api-docs\conf.py` Qt section referencing `InvestModelInput`?
> * Do we need `exe/qt.conf` anymore?
> 
> Otherwise this is looking good!

Thanks @dcdenu4 , I believe you're correct. I removed all those suggestions and updated the `cli.rst`

dcdenu4 commented:
Thanks @davemfish, looks like some merge conflicts popped up... which seems strange for the `ui` files.

dcdenu4 commented on doc/api-docs/cli.rst:
```diff
@@ -22,50 +22,52 @@ entry point::
+                            Taskgraph prints to the console and (if running in headless
                             mode) how much is written to the logfile.
-      --debug               Enable debug logging. Alias for -vvvvv
+      -L {en,es,zh}, --language {en,es,zh}
```
Thanks for these updates Dave!

dcdenu4 commented on exe/qt.conf:
```diff
@@ -1,4 +0,0 @@
-# solves problem with binaries not launching on mac.
```
Looks like this might still be used... although I thought I grepped for it. From actions:

>150595 INFO: Appending 'datas' from .spec
Unable to find "D:\a\invest\invest\exe\qt.conf" when adding binary and data files.

dcdenu4 replied:
Looks likes it's in `exe/invest.spec`

>     'datas': [('qt.conf', '.'), proj_datas],

dcdenu4 reviewed (changes requested):
Hey @davemfish, along with the merge conflicts it looks like `exe/qt.conf` is still being used by something.

dcdenu4 reviewed (approved):
Thanks @davemfish!
